src/services/typeService.ts

```diff
@@ -0,0 +1,133 @@
+        const user = stringToObjectId('123456789012');
+
+        if (!user) {
+            return { error: 'user not found' };
+        }
+
+        const types = await TypeModel.find<IType>({ user });
+
+        return {
+            types,
+        };
+    } catch (error) {
+        return { error };
+    }
+}
+
+export async function createType(name: string, isDefault = false) {
+    try {
+        await connectDB();
+        // TODO: Get USER_ID from session
+        const user = stringToObjectId('123456789012');
+
+        if (!user) {
+            return { error: 'user not found' };
+        }
+
+        const type:IType = await TypeModel.create({ name, user, isDefault });
+
+        return {
+            type,
+        };
+    } catch (error) {
+        return { error };
+    }
+}
+
+export async function getType(id: string) {
+    try {
+        await connectDB();
+        // TODO: Get USER_ID from session
+        const user = stringToObjectId('123456789012');
+
+        if (!user) {
+            return { error: 'user not found' };
+        }
+
+        const parsedId = stringToObjectId(id);
+
+        if (!parsedId) {
+            return { error: 'Type not found' };
+        }
+        const type = await TypeModel.findOne<IType>({ _id: parsedId, user });
+        if (type) {
+            return {
+                type,
+            };
+        } else {
+            return { error: 'Type not found' };
+        }
+    } catch (error) {
+        return { error };
+    }
+}
+
+export async function updateType(id: string, { name }: { name?: string }) {
+    try {
+        await connectDB();
+        // TODO: Get USER_ID from session
+        const user = stringToObjectId('123456789012');
+
+        if (!user) {
+            return { error: 'user not found' };
+        }
+
+        const parsedId = stringToObjectId(id);
+
+        if (!parsedId) {
+            return { error: 'Type not found' };
+        }
+
+        const type = await TypeModel.findOneAndUpdate<IType>(
+            { _id: parsedId, user },
+            { name },
+            { new: true }
+        );
+
+        if (type) {
+            return {
+                type,
+            };
+        } else {
+            return { error: 'Type not found' };
+        }
+    } catch (error) {
+        return { error };
+    }
+}
+
+export async function deleteType(id: string) {
+    try {
+        await connectDB();
+        // TODO: Get USER_ID from session
+        const user = stringToObjectId('123456789012');
+
+        if (!user) {
+            return { error: 'user not found' };
+        }
+
+        const parsedId = stringToObjectId(id);
+
+        if (!parsedId) {
+            return { error: 'Type not found' };
+        }
+        const type = await TypeModel.findOneAndDelete<IType>({ _id: parsedId, user, isDefault: false });
+        if (type) {
+            return {};
+        } else {
+            return { error: 'Type not found' };
```

Comment:
[NIT] I think it's better not to give false information

Reply:
Updated.